ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/ShelleyHFC.hs

```diff
@@ -153,11 +153,13 @@ shelleyTransition ShelleyPartialLedgerConfig{..}
 
     isTransition :: ShelleyLedgerUpdate era -> Maybe EpochNo
     isTransition (ShelleyUpdatedPParams maybePParams newPParamsEpochNo) = do
-         pp <- maybePParams
+         -- We need to run this check first in order to ensure that PParams value is not
+         -- forced unnecessarily. See documentation of `futurePParamsGovStateG` in ledger.
+         guard $ shelleyAfterVoting >= fromIntegral k
+         SL.SJust pp <- Just maybePParams
```

Comment:
FTR: We just discussed in a call that this is no longer necessary, `nextEpochUpdatedPParams` is always cheap (by returning `SNothing` when the proposals for the next epoch are not yet definitive)

Reply:
Yep. This check is now removed.